aten/src/ATen/native/cuda/PowKernel.cu

```diff
@@ -140,9 +141,10 @@ void pow_tensor_scalar_kernel(TensorIteratorBase& iter, const Scalar& exp_scalar
     }
   }
   if (isComplexType(iter.common_dtype()) || exp_scalar.isComplex()) {
-    AT_DISPATCH_COMPLEX_TYPES(iter.common_dtype(), "pow_cuda", [&]() {
-      const auto exp = exp_scalar.to<scalar_t>();
-      gpu_kernel(iter, [=]GPU_LAMBDA(scalar_t base) -> scalar_t {
+    AT_DISPATCH_COMPLEX_TYPES_AND(kComplexHalf, iter.common_dtype(), "pow_cuda", [&]() {
+      using opmath_t = at::opmath_type<scalar_t>;
+      const auto exp = exp_scalar.to<opmath_t>();
+      gpu_kernel(iter, [=]GPU_LAMBDA(opmath_t base) -> opmath_t {
```

Comment:
again, lambda arg should be scalar_t, not opmath_t, cast should happen inside lambda function